src/Reducer.js

```diff
@@ -86,10 +84,18 @@ function update(state,action){
             el.children[ind] = getInitialState(newProps, newState.scenes, ind, action);
             return newState;
 
+        case PUSH_TO_CURRENT_ACTION:
+            parent = getCurrent(newState).parent;
+            newProps.parent = parent;
+            el = findElement(newState, parent);
+            assert(el, "Cannot find element for parent="+parent+" within current state:"+JSON.stringify(newState));
+            // fall through to PUSH_ACTION
+
         case PUSH_ACTION:
             el.children.push(getInitialState(newProps, newState.scenes, el.children.length, action));
             el.index = el.children.length - 1;
-            newState.scenes.current = getCurrent(newState);
+            newState.scenes.current = getCurrent(newState).key;
+            newState.scenes[newState.scenes.current] = newProps;
```

Comment:
Modification of scenes could bring some unpleasant issues - it is possible that we could have several scenes in stack with the same sceneKey (such issue was created already in the past), for example on Details of user1 and then Details of user2. And you are proposing to set 'scene' with user1 or user2?